src/Jackalope/Transport/DoctrineDBAL/Query/QOMWalker.php

```diff
@@ -323,18 +332,21 @@ public function walkJoinSource(QOM\JoinInterface $source)
         $sql .= "AND " . $this->walkJoinCondition($source->getLeft(), $source->getRight(), $source->getJoinCondition()) . " ";
         $sql .= ") "; // close on-clause
 
-        $sql .= "WHERE $leftAlias.workspace_name = ? AND $leftAlias.type IN ('" . $source->getLeft()->getNodeTypeName() ."'";
-        $subTypes = $this->nodeTypeManager->getSubtypes($source->getLeft()->getNodeTypeName());
-        foreach ($subTypes as $subType) {
-            /* @var $subType \PHPCR\NodeType\NodeTypeInterface */
-            $sql .= ", '" . $subType->getName() . "'";
+
+        if ($root) {
```

Comment:
@dantleech is there an easy solution here? can we just add the type query for each part instead of only the leftmost?
if there is no obvious solution, then yes lets add a TODO, its the same behaviour we currently had, which seems unfinished but as long as everybody uses nt:unstructured everywhere is irrelevant anyways.